homework-classes/App.js

```diff
@@ -19,12 +19,24 @@ class App {
 
     const root = document.getElementById('root');
     const header = Util.createAndAppend('header', root, { class: 'header' });
+    Util.createAndAppend('h1', header, { text: 'HYF Repositories' });
     this.mainContainer = Util.createAndAppend('div', root, { id: 'container' });
 
     try {
       const repos = await Util.fetchJSON(url);
       this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      const select = Util.createAndAppend('select', header);
+      this.repos.sort((a, b) => a.name().localeCompare(b.name()));
+      this.repos.forEach((repo, index) => {
+        Util.createAndAppend('option', select, {
+          text: repo.repository.name,
```

Comment:
In OOP, the data from a class (in this case the Repository class), should be considered private to that class. If another class, such as in this case the App class, it should request the Repository class to provide that data, by calling one of its methods. This principle is called 'data encapsulation' (google for it). As you can see in the Repository class, a method was provided for this purpose, to obtain the name. (You di it correctly in line 29).

So this line should read:

```js
text: repo.name(),
```